### PR TITLE
Added SendGrid Documentation

### PR DESCRIPTION
**Pull request by shahednasser**
Added SendGrid documentation. This depends on #1293 because it changes the sidebar.

### CODE REVIEW

File: docs/content/add-plugins/sendgrid.mdx [hunk 1]
```diff
@@ -0,0 +1,3967 @@
...
+
+If you don’t have a SendGrid account, make sure to [create one](https://signup.sendgrid.com) first. You also need to set up a [single sender](https://docs.sendgrid.com/ui/sending-email/sender-verification) first in SendGrid before you can start sending emails.
+
+### API Key
+
+For the integration to work, you need to create an API key in your SendGrid account.
+
+You can do that by choosing from the sidebar of your SendGrid dashboard Settings > API Keys. Then, click on Create API Key.
+
+:::note
```
Comment (srindom) [inline docs/content/add-plugins/sendgrid.mdx:39]: question(non-blockin): Is this supposed to be here?

**Review by srindom — APPROVED** [review]
LGTM!

**Comment by srindom** [comment]
@shahednasser feel free to merge :)

File: docs/content/add-plugins/sendgrid.mdx (hunk 1 above)
Comment (shahednasser) [reply docs/content/add-plugins/sendgrid.mdx:39]: Yes this adds the note formatting in docusaurus

Screen Shot 2022-04-12 at 10 03 51 PM

**Pull request merged**